src/index.ts

```diff
@@ -120,9 +145,61 @@ const GlobalIdExtensionsTweakFieldsPlugin: Plugin = function(builder) {
       });
     }, fields);
   });
+
+  // add deprecations
+  builder.hook("GraphQLObjectType:fields:field", (field, { inflection }, context) => {
+    const {
+      scope: { pgFieldIntrospection }
+    } = context;
+
+    if (
+      !pgFieldIntrospection ||
+      !pgFieldIntrospection.class ||
+      pgFieldIntrospection.kind !== "attribute"
+    ) {
+      return field;
+    }
+
+    const attr: PgAttribute = pgFieldIntrospection;
+    const table: PgClass = pgFieldIntrospection.class;
+
+    if (containsSingleColumn(table.primaryKeyConstraint, attr)) {
+      return maybeDeprecate(field, attr, 'nodeId');
+    }
+
+    const fk = table.constraints.find(c =>
+      isForeignKey(c) &&
+      containsSingleColumn(c, attr)
+    );
+
+    if (fk) {
+      const fieldName = inflection.singleRelationByKeys(
+        fk.keyAttributes,
+        (fk as any).foreignClass,
+        table,
+        fk
+      );
+
+      return maybeDeprecate(field, attr, `${fieldName}.nodeId`);
+    }
+
+    return field;
+  });
+
+  function maybeDeprecate<T extends GraphQLFieldConfig<any, any>>(field: T, attr: PgAttribute, preferredField: string): T {
+    const condition = typeof options.globalIdShouldDeprecate === 'function'
+      ? options.globalIdShouldDeprecate(attr)
+      : options.globalIdShouldDeprecate;
+
+    const deprecationReason = typeof options.globalIdDeprecationReason === 'function'
+      ? options.globalIdDeprecationReason(preferredField)
+      : options.globalIdDeprecationReason;
+
+    return condition ? { ...field, deprecationReason } : field;
```

Comment:
I suppose it's possible people could return a null deprecation reason:

```
    return condition && deprecationReason ? { ...field, deprecationReason } : field;
```

Also, if there's an existing deprecation reason we should probably go with that rather than adding our own?

```suggestion
    return condition && deprecationReason ? { ...field, deprecationReason: field.deprecationReason || deprecationReason } : field;
```